src/ucp/core/ucp_worker.c

```diff
@@ -2258,3 +2280,206 @@ void ucp_worker_print_info(ucp_worker_h worker, FILE *stream)
 
     UCP_WORKER_THREAD_CS_EXIT_CONDITIONAL(worker);
 }
+
+static unsigned ucp_worker_discard_uct_ep_destroy_progress(void *arg)
+{
+    ucp_request_t *req  = (ucp_request_t*)arg;
+    uct_ep_h uct_ep     = req->send.discard_uct_ep.uct_ep;
+    ucp_worker_h worker = req->send.discard_uct_ep.ucp_worker;
+    khiter_t iter;
+
+    ucp_trace_req(req, "destroy uct_ep=%p", uct_ep);
+    ucp_request_put(req);
+
+    UCS_ASYNC_BLOCK(&worker->async);
+    --worker->flush_ops_count;
+    iter = kh_get(ucp_worker_discard_uct_ep_hash,
+                  &worker->discard_uct_ep_hash, uct_ep);
+    if (iter == kh_end(&worker->discard_uct_ep_hash)) {
+        ucs_fatal("no %p UCT EP in the %p worker hash of discarded UCT EPs",
+                  uct_ep, worker);
+    }
+    kh_del(ucp_worker_discard_uct_ep_hash,
+           &worker->discard_uct_ep_hash, iter);
+    UCS_ASYNC_UNBLOCK(&worker->async);
+
```

Comment:
shall we unprogress UCT iface like we added in https://github.com/openucx/ucx/commit/582c6b41a35a5edb22e306de0d2e5e488ad1c660?

Reply:
I'd port the latest available version to int3 to avoid missing some fixes/improvements

Reply:
OK. I'll port https://github.com/openucx/ucx/pull/7819 to this PR.

Reply:
> OK. I'll port [openucx#7819](https://github.com/openucx/ucx/pull/7819) to this PR.

@yosefe @dmitrygx  [openucx#7819](https://github.com/openucx/ucx/pull/7819) was created to resolve performance issue when using TCP. Do we really need to port it to the int3 branch in which TCP is not supported?

Reply:
I guess deactivating progress functions for other transports should be done too.
Btw, it worth porting other PRs which were added after mering he main PR which introduces "discard", .e.g:
- https://github.com/openucx/ucx/pull/6401
- https://github.com/openucx/ucx/pull/6933
- https://github.com/openucx/ucx/pull/5712
- https://github.com/openucx/ucx/pull/6605
- https://github.com/openucx/ucx/pull/5882
and I'd check that the current state of all discarding functions in the PR are aligned with ones from the master branch.

Reply:
@dmitrygx Thanks for find out all the related PRs! Since the int3 branch is used by Tencent and they want to keep it stable, so minimal changes is preferred. If all above PRs are must, then let's port them.

Reply:
@yosefe @dmitrygx @changchengx So let's don't port https://github.com/openucx/ucx/pull/7819, but port all other discarding related PRs listed by Dmitry. WDYT?

Reply:
No need to port them as well, since that branch does not need to support multi-device.

Reply:
> No need to port them as well, since that branch does not need to support multi-device.

but it is not about mu;ti-device, it some bugfixes implemented on top of the initial simple,entation

Reply:
@dmitrygx I'll check the patches listed you.
Do you know what test case could trigger the fixed problem in the listed patches?
After porting the necessary patches, I want to test them if it's possible.

Reply:
> 
> but it is not about mu;ti-device, it some bugfixes implemented on top of the initial simple,entation
>
@yosefe What do you think below PRs listed by Dmitry? Do you agree with porting them to integration3? It changes about 600 Line-of-Code 
https://github.com/openucx/ucx/pull/6401
https://github.com/openucx/ucx/pull/6933
https://github.com/openucx/ucx/pull/5712
https://github.com/openucx/ucx/pull/6605
https://github.com/openucx/ucx/pull/5882